corehq/apps/reports/standard/cases/basic.py

```diff
@@ -24,6 +26,129 @@
 from .data_sources import CaseInfo, CaseDisplay
 
 
+class CaseExportScope(object):
```

Comment:
What are the changes in logic?  The problem with doing it like this is that it's really difficult to tell how the behavior changes between for instance, the Case List Report and the Reassign Cases Report (since they're supposed to be the same).  Is there another way to fork the logic on `can_access_all_locations`?  It's safe to assume that you can change any behavior when `can_access_all_locations == False`, as long as the behavior is consistent when it's `True`.